cpp/ql/lib/semmle/code/cpp/ir/dataflow/internal/DataFlowUtil.qll

```diff
@@ -1533,6 +1552,12 @@ private class IndirectArgumentOutExprNode extends ExprNodeBase, IndirectArgument
   final override Expr getConvertedExpr(int n) { exprNodeShouldBeIndirectOutNode(this, result, n) }
 }
 
+private class IndirectTemporaryExpr extends ExprNodeBase instanceof IndirectOperand {
+  IndirectTemporaryExpr() { exprNodeShouldBeIndirectOperand(this, _, _) }
```

Comment:
I feel like there's a mismatch between this class, which claims that the node is a temporary expr, with the QLDoc on `exprNodeShouldBeIndirectOperand`, which only claims the node is an `IndirectOperand` (that should have a `.asExpr()`).  Its probably just be me, missing a connection here.

Reply:
That's fair. I'll rename the class to something more generic. It's currently true because `exprNodeShouldBeIndirectOperand` restricts the node to be an indirection of an operand whose definition is related to an temporary IR variable. But that's not guaranteed to be true if we find other uses of this class.

If it's okay with you, I'll do this in a subsequent PR so that you can get this change into https://github.com/github/codeql/pull/15371 without waiting for another round of CI